### ISSUE TITLE
Replay system and replay video capture

### ISSUE THREAD

**Issue by Richard-L** [stenzek/duckstation#752]
Not really an "issue", more a discussion no how much this interests you Sten.

For competitive gameplay and associated community video content (youtube, etc), recording one's "performance" in the emulator as a replay file first, and then into video, is of interest. Most arcade emulators I know offer a replay feature. Do you think this is possible with PS1 emulation, and if, is it something that interests you?

Further, how realistic is a video capture of (a) such recorded gameplay, or (b) even live footage?

I suppose for latter case there are lots of secondary apps like OBS, ShareX or similar. Even just for error reporting though, I have found myself wishing for a capture option in DS.

**Comment by jpchow26** [comment]
the Xbox game bar (Win+G) on Windows 10 is a built in OS utility for capturing. Works well

**Comment by Richard-L** [comment]
Yeah just not everyone is on Win10, and on mine the game bar doesn't work at all.

**Comment by MojoJojoDojo** [comment]
So TAS support basically?

**Comment by Richard-L** [comment]
I guess that goes along with it since you can edit the replay? Don't know the TAS world too well.

**Comment by IPeluchito** [comment]
Would this work?
https://github.com/baSSiLL/SharpAvi

**Comment by Richard-L** [comment]
@IPeluchito raw avi? Afraid the dumping of it is heavier than realtime encoding to a modern compression standard.

Didn't know they even still used the avi container.

**Comment by BCGamerYT** [comment]
> So TAS support basically?

Also TASVideos or TAS Studio?

**Comment by stenzek** [comment]
Audio/video capture added in https://github.com/stenzek/duckstation/commit/af47eb6956d30e8a01cd3f4828132e97e28e41a8.

Currently Media Foundation/Windows only, I'll add ffmpeg at some point.
